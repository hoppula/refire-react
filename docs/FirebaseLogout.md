# FirebaseLogout

> Logs user out of Firebase when given child component is clicked.

```js
import { FirebaseLogout } from 'refire-react'
const LogoutButton = () => {
  return (
    <FirebaseLogout>
      <Button>Logout</Button>
    </FirebaseLogout>
  )
}
```
