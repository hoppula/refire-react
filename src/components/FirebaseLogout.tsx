import * as React from "react"
import { Component } from "react"
import * as ReactRedux from "react-redux"
import { connect } from "react-redux"
import { firebaseActions } from "refire"

const { logout } = firebaseActions

export interface LogoutProps {
  readonly children?: any,
  readonly dispatch?: ReactRedux.Dispatch<any>
}

export class FirebaseLogout extends Component<LogoutProps & ReactRedux.DispatchProp<any>, {}> {

  logout = () => {
    this.props.dispatch(logout())
  }

  render() {
    return React.Children.only(
      React.cloneElement(this.props.children, {
        onClick: this.logout
      })
    )
  }

}

export default connect()(FirebaseLogout)
