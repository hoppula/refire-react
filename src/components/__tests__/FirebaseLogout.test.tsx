import * as React from "react"
import * as renderer from "react-test-renderer"

import { FirebaseLogout } from "../FirebaseLogout"

it("FirebaseLogout renders given children and calls dispatch on click", () => {
  const logoutFn = jest.fn()

  let component = renderer.create(
    <FirebaseLogout dispatch={ logoutFn }>
      <button>Logout</button>
    </FirebaseLogout>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()
  expect(logoutFn.mock.calls.length).toBe(1)
})
