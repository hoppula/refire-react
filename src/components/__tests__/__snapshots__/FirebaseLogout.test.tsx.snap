// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirebaseLogout renders given children and calls dispatch on click 1`] = `
<button
  onClick={[Function]}
>
  Logout
</button>
`;
